### DISCUSSION TITLE
Delay in Real-Time Monitoring Graphs Compared to Database Time

### DISCUSSION THREAD

**Discussion by vishalkadam18** [apache/skywalking#12322] [Q&A]
Dear Team, 

We've been actively monitoring our Linux infrastructure using Apache SkyWalking for the past few days. However, we've encountered an issue where the graphs displayed in the monitoring dashboard are consistently delayed by approximately 10 to 12 minutes compared to the time shown in the database. As a result, we're unable to monitor real-time data effectively.

Despite the database providing the exact time and data, the discrepancy between the real-time graphs and database time is causing challenges in our monitoring process.

![Screenshot from 2024-06-11 16-50-36](https://github.com/apache/skywalking/assets/115125925/c0f62cb1-a685-40bf-a466-d6ebe367302b)
![Screenshot from 2024-06-11 16-49-09](https://github.com/apache/skywalking/assets/115125925/07f4f649-97cb-42f7-a4c9-527dd0fdf132) 

Regards & Thanks,
Vishal Kadam

**Comment by wu-sheng** [comment]
I am not sure about `approximately 10 to 12 minutes`. The time visualized on the UI are only calculated based on your local timezone.

The last menu item is about settings including this.
<img width="653" alt="image" src="https://github.com/apache/skywalking/assets/5441976/0da0ea25-0274-49be-a255-9b3dd8572723">

You could check the GraphQL query(through Browser logs) to verify the actual response. 

From my understanding, if the data is correct in the database, all things should be right. The only effect factor should be the UI side process.

**Reply by vishalkadam18** [reply]
Can I confirm whether the time_bucket field is included when fetching with values from a table?

As per my understanding i am sure that from db only is value is taking not time_bucket.

**Reply by wu-sheng** [reply]
Yes, you could find that from the time series value query.

<img width="471" alt="image" src="https://github.com/apache/skywalking/assets/5441976/7b292906-14c0-4ede-b2db-e9eda0562632">

The name(ID) actually is the `ms` timestamp. `1718283240000` is `2024-06-13 20:54:00`, which should match the time bucket in the database.

**Reply by vishalkadam18** [reply]
> Yes, you could find that from the time series value query.
> 
> <img alt="image" width="471" src="https://private-user-images.githubusercontent.com/5441976/339377371-7b292906-14c0-4ede-b2db-e9eda0562632.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3MDI5MjUsIm5iZiI6MTcxODcwMjYyNSwicGF0aCI6Ii81NDQxOTc2LzMzOTM3NzM3MS03YjI5MjkwNi0xNGMwLTRlZGUtYjJkYi1lOWVkYTA1NjI2MzIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MThUMDkyMzQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjIxMzNhMGI3ZmM0ODIxZGJjMjY1N2M1OTgyOTNmMDQ2YjA2MmViOWI1YmMyZTMyYTViMTEwODhlN2Y3YzdhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.5pJq7gCpr_bUyikGm33RniykuapzUD9nm0QBaVFFCWM">
> The name(ID) actually is the `ms` timestamp. `1718283240000` is `2024-06-13 20:54:00`, which should match the time bucket in the database.

Based on my understanding, the problem does not seem to be at the UI level. Instead, it appears to stem from delays in retrieving data from the database. I have tried version 10.0.0, but the issue persists. Could you please guide me on how to resolve this query-related issue?

**Reply by wu-sheng** [reply]
I can't see any delay from my end. It should be 30s to 1 mins delayed, as we don't that intentionally ckr metrics. Much less delay for logs and tracea.
